modules/cli/src/main/scala/scala/cli/commands/Compile.scala

```diff
@@ -52,7 +53,13 @@ object Compile extends ScalaCommand[CompileOptions] {
       }
       else if (options.classPath)
         for (s <- builds.main.successfulOpt) {
-          val cp = s.fullClassPath.map(_.toAbsolutePath.toString).mkString(File.pathSeparator)
+          val testBuildOpt = builds.get(Scope.Test)
+          val testFullClassPath =
+            testBuildOpt.flatMap(_.successfulOpt).map(_.fullClassPath).getOrElse(Nil)
+          val fullClassPath = s.fullClassPath.union(testFullClassPath).distinct
```

Comment:
I'm not sure about the `union`… Keeping either the main class path or the test one should be enough. Something like this a few lines above should do:
```scala
for (s <- builds.get(Scope.Test).orElse(builds.main.successfulOpt))
```